src/ajax.js

```diff
@@ -333,7 +349,15 @@ jQuery.extend({
 			"text html": true,
 
 			// Evaluate text as a json expression
-			"text json": jQuery.parseJSON,
+			"text json": function(json) {
```

Comment:
Instead of modifying the standard `json` converter, you could create a `jsonAsync`. That way we don't change behavior for existing code.